packages/legend-query/src/stores/QueryBuilderLambdaBuilder.ts

```diff
@@ -209,6 +221,66 @@ export const processPostFilterOnLambda = (
   return lambda;
 };
 
+export const buildPropertyExpressionChain = (
```

Comment:
It's safe to make the param `propertyExpression` of type `AbstractPropertyExpression`. And as such, we can make this method return `AbstractPropertyExpression` and eliminate the awkward check and `undefined` return.